### PR TITLE
Remove unused node server and cleanup deps

### PR DESCRIPTION
**Pull request by limemloh**
## Purpose

The repository contains a lot of unneeded files and dependencies.

## Changes

- Remove node server and instead use `webpack-dev-server` which is already a dependency.
- Remove unused dependencies.
- Update the remaining dependencies.
- Cleanup webpack build config.
- Remove unused grpc files and documentation.
- Remove unused scripts in package.json.
- Remove unsued shell scripts.
- Update the Node version build dependency.
- Update the k8s.Dockerfile

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [ ] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,120 +4,39 @@
...
-    "build-client": "NODE_ENV=production webpack -p && cp -rp assets dist/public",
-    "dev-client": "webpack-dev-server -w",
-    "dev-client:hot": "webpack-dev-server -w --hot",
-    "build-server": "tsc -p ./src/server",
-    "dev-server": "tsc-watch -p ./src/server --onSuccess \"node dashboard-backend.js\"",
-    "dev": "npm run clean && concurrently --prefix \"[{name}]\" --names \"server,client\" -c \"blue,magenta\" \"npm run dev-server\" \"npm run dev-client\"",
-    "dev:hot": "npm run clean && concurrently --prefix \"[{name}]\" --names \"server,client\" -c \"blue,magenta\" \"npm run dev-server\" \"npm run dev-client:hot\"",
-    "build": "npm run clean && concurrently --prefix \"[{name}]\" --names \"server,client\" -c \"blue,magenta\" \"npm run build-server\" \"npm run build-client\""
+    "formatcheck": "elm-format src/elm/ --validate",
+    "build": "npm run clean && NODE_ENV=production webpack && cp -rp public/* dist",
```
Comment (bisgardo) [inline package.json:13]: It looks like webpack should be able to do copy like this as part of the build process: https://stackoverflow.com/questions/27639005/how-to-copy-static-files-to-build-directory-with-webpack

**Review by bisgardo — APPROVED** [review]
Looks good!

Except for the favicons and `elm-pep.js`, none of the asset files actually seem to be used anymore (also not really sure what the latter is for).

All the comments should be considered suggestions and obervations - feel free to ignore what you consider irrelevant.

File: webpack.config.js [hunk 2]
```diff
@@ -1,78 +1,57 @@
...
+      template: './src/index.ejs',
+      templateParameters: {
+        title: 'Concordium Dashboard',
+        analyticsId,
+      },
+    }),
+    new webpack.DefinePlugin({
+      __VERSION__: JSON.stringify(packageJson.version),
+      __ANALYTICS_ID__: JSON.stringify(analyticsId),
+      __PRODUCTION__: isProduction.toString(),
```
[the remark this answers is not in the document]
Comment (limemloh) [reply webpack.config.js:48]: DefinePlugin will replace `__PRODUCTION__` with the content of the string, which is also why the above have quotes as part of the string (done by stringify).
They do fall back to call toString for anything else, but I prefer to be explicit and call it manually.

File: k8s.Dockerfile [hunk 3]
```diff
@@ -1,24 +1,20 @@
...
+FROM node:14 as build
 
 # https://github.com/nodejs/docker-node/blob/master/docs/BestPractices.md
 ENV NPM_CONFIG_PREFIX=/home/node/.npm-global
 # USER node
 
 WORKDIR /home/node/app
 
-# Possibly temporary, see proto/readme.md
 COPY . .
```
[the remark this answers is not in the document]
Comment (limemloh) [reply k8s.Dockerfile:9]: yes, I'll update this

File: k8s.Dockerfile [hunk 4]
```diff
@@ -1,24 +1,20 @@
...
-RUN npm install concurrently
+# Unsafe permission is needed for installing elm through npm
+RUN npm install --unsafe-perm=true
 RUN npm run build
 
 FROM nginx:alpine
 
-COPY --from=build /home/node/app/dist/public /usr/share/nginx/html/
+COPY --from=build /home/node/app/dist/ /usr/share/nginx/html/
 RUN sed -i 's/index.*$/try_files \$uri \/index.html =404;/g' /etc/nginx/conf.d/default.conf
```
[the remark this answers is not in the document]
Comment (limemloh) [reply k8s.Dockerfile:18]: yes, I agree, I'll copy the file instead

**Comment by limemloh** [comment]
I have remove `elm-pep.js` completely since pointer events is supported by most browsers https://caniuse.com/pointer

**Pull request merged**